### ISSUE TITLE
How to hide passed tests?

### ISSUE THREAD

**Issue by RomanPerin** [trentmwillis/ember-exam#42]
Hi team!

Can I use some config file (e.g. testem.js... or another..) to hide passed tests?
It will be very helpful.

Thx in advance.

**Comment by trentmwillis** [comment]
`ember exam` just extends the default `ember test` command, so you can still hide passed tests just as you normally would (by using `&hidepassed`).

**Comment by trentmwillis** [comment]
Going to close this, let me know if the above is not what you were looking for.

**Comment by RomanPerin** [comment]
Hi trentmwillis,

Actually, I'm looking for another solution...
I have 'testem.json' file in my project. It contains next lines:
{
  "framework": "qunit",
  "test_page": "tests/index.html?hidepassed&coverage",
  "disable_watching": true,
  "launch_in_dev": [
    "Firefox",
    "Safari"
  ]
}
Here I have next option:
"test_page": "tests/index.html?hidepassed&coverage",
So, 'hidepassed' query is set by default when I run tests as usual.
But when I run tests with 'ember-exam' addon, 'hidepassed' option is omitted by default, so I have to write it with the 'ember exam' command in the console each time when I run tests.

So, I want to have config-file (testem.json or some other) to be able to configure default options for the 'ember exam' command.
Is it possible?

**Comment by trentmwillis** [comment]
Ah okay, I see what you're saying. After looking into it briefly it seems like the issue is a conflict between setting [`test_page` via the CLI options ](https://github.com/testem/testem/blob/master/docs/config_file.md#cli-level-options) and setting it via the config file.

In Ember Exam we set up the test page programmatically which I think overrides the config file value. We'll probably need to read in the config file to append the values.

**Comment by RomanPerin** [comment]
Yes, it is exactly want I want.
It will be great to see this fix in the nearest future)

Thank you for your time.

**Comment by trentmwillis** [comment]
@RomanPerin can you give me some steps to reproduce what you're seeing? As far as I can tell, `hidepassed` should get applied in the URL, but it looks like Testem is doing some weird thing that adds an `=` after it. E.g., `tests/index.html?hidepassed=&_split=2`

**Comment by RomanPerin** [comment]
I can be wrong, but it seems like you ignore testem.js file and define default settings in the code..
For example ember-exam/lib/commands/exam.js:_generateCustomConfigs function 
has next code: 
 ```
var baseUrl = config.testPage || 'tests/index.html?hidepassed';
var splitUrl = addToUrl(baseUrl, '_split', count);
```
So, it seems like addToUrl function do not work correct for this case.

**Comment by trentmwillis** [comment]
Okay, so this only happens when using the `parallel` option then. I had incorrectly assumed `config.testPage` would contain the value from the `testem.json`, but that is not the case.

Working on a fix. Would still appreciate some reproduction steps so I can verify that I fix your issue.

**Comment by RomanPerin** [comment]
Thx)
Look, I use next command:
```
ember exam --split 8 --parallel --server
```
And no matter what settings I have in a testem.json file, it runs with a wrong query like:
```
tests/index.html?hidepassed=&_split=2
```
So it has wrong generated 'hidepassed' query param.

**Comment by trentmwillis** [comment]
Turns out this was actually a two-fold bug:

1. `ember-exam` wasn't using the `test_page` value provided in the Testem config. This should be fixed via #52 and released in version `0.4.6`.
2. Testem didn't properly handle merging query params when they weren't key-value pairs. This is fixed by testem/testem#946 and released in `testem@1.10.3`.

So, this is fixed now, but you'll need to update both ember-exam and testem (the latter is a transitive dependency of Ember-CLI, so you might just try re-installing Ember-CLI in your project).
